### ISSUE TITLE
Text replaced by symbol in terminal.

### ISSUE THREAD

**Issue by AXington** [mintty/wsltty#173]
I am trying to figure out how to disable this feature, but in several places, text like `fi` replaced is replaced by a phone symbol. This obviously gets really really frustrating. I can't find anywhere to disable this.

![fi_phone](https://user-images.githubusercontent.com/2721182/60280377-cf4c1480-98d0-11e9-8700-93a4ffa130f3.JPG)

**Comment by mintty** [comment]
Please provide information what you are actually writing to the terminal.
If `ls > lsfile.txt ; cat lsfile.txt` produces the same effect, upload the lsfile.txt here. Otherwise, try to generate a file that can be used as a test case; find what parameters are set up for `ls`, or make a terminal log.

**Comment by AXington** [comment]
[lsfile.txt](https://github.com/mintty/wsltty/files/3335686/lsfile.txt)
![lsfile](https://user-images.githubusercontent.com/2721182/60282585-4edbe280-98d5-11e9-8fd1-bbc7a2983da2.JPG)


I'm also looking into why directory names are highlighted in green. Seems to be a weird issue with zsh-syntax-highlighting, though, I don't think that's related. May open another issue for that if I can't figure it out.

**Comment by Biswa96** [comment]
* What `zsh` theme are you using?
* Did you add any prompt text? Output of `echo $PS1`
* Does `wsl.exe` shows it correctly?

**Comment by AXington** [comment]
zsh theme: agnoster

``` echo $PS1
%{%f%b%k%}$(build_prompt)
```

wsl.exe shows it correctly, and with certain fonts, it renders the theme much better, which leads to my next discovery... 

This whole thing just got a lot more interesting too... The font I use for all my terminals (iterm2, konsole, alacritty, wsltty) across multiple OS's doesn't show in wsl.exe, it also doesn't have that issue in any of my other terminals. So I switched to a different font in it (the default didn't have my powerline patches), I tried selecting the same font but it's not on the list. So then to make sure I was showing the same fonts/etc on wsltty, the one I selected in wsl.exe was on the list, but when I pressed 'OK' it said there was an error dialog (see screenshot). Though, I was able to hit 'Apply' then 'Cancel' and it applied the font. 
![font_error](https://user-images.githubusercontent.com/2721182/60284004-7f714b80-98d8-11e9-88d7-847e5146b05e.JPG)

After applying the new fonts, however, the problem went away. Though, I think I found a new font I like better than the one I was using that doesn't have this issue... 

I was using the meslo for powerline family, and I tried with every available version.
https://github.com/powerline/fonts



Also one more screenshot for increased weirdness...
![echo_phone](https://user-images.githubusercontent.com/2721182/60283521-54d2c300-98d7-11e9-9933-5ddf54a07595.JPG)

I guess you can close this issue if you want, but it's really strange.

**Comment by mintty** [comment]
The file contains nothing special, it's simply the sequence "fi" that's getting rendered as a phone.
Which quite certainly means that you were using a font that supports ligatures, and that the "fi" ligature in that font displays as a phone. Which in turn means the font is broken. It's a font bug.

About zsh input colouring, that is not a terminal issue; the terminal simply produces the text attributes as requested via escape sequences.

**Comment by mintty** [comment]
It does not happen with one of the Meslo fonts I just downloaded from there.
Note there is a commit notice that particularly mentions "fi" and "fl" ligatures.

**Comment by AXington** [comment]
Yeah, I knew the file itself wouldn't contain anything strange, but I wanted to make sure to give you any and all information you wanted and or seemed relevant.

What's strange is, I did reinstall the fonts during the troubleshooting process I did above, from a brand new clone, using the powershell script in the repo, had the same problem, but the font I chose renders a heck of a lot better too. There seemed to be something off about the anti-aliasing or 'font smoothing' on the Meslo font that doesn't seem to be a problem with the new font I chose (noto). I'm going to chalk this up to a combination of issues with a bug in that font and something weird with Windows.

Windows hasn't been my main OS for years, so I'm admittedly quick to blame it for things that, honestly, aren't always fair, but the way it handles things like fonts vs Linux/BSD/Mac seems odd and less stable to me...
